### DISCUSSION TITLE
Dirichlet inflow boundary conditions

### DISCUSSION THREAD

**Discussion by TRPrasanna** [loganoz/horses3d#197] [Q&A]
Hi, I was wondering if the proper way to fix the inflow boundary conditions to a fixed value (say, values corresponding to boundary layer profiles) would be to do this every time step via the UserDefinedPeriodicOperation function in the problem setup file. However, the thermodynamic variables are not passed to the aforementioned function. Is there a more elegant way to do this?

Edit: I just realized that the intended way to do this is by setting the boundary condition to User-defined in the control fine. Then, would the correct way to implement the user-defined BC in the problem file be using UserDefinedState1, which is called once for all the spatial coordinates every time step in the chosen boundary?

Another edit: I have seem to run across an issue in UserDefinedState1 : I need access to the time step size (dt) within this subroutine and I can't see a straightforward way to do so.

**Comment by loganoz** [comment]
Hi @TRPrasanna,

You are advancing in the right direction. The correct way to implement Dirichlet boundary conditions is with the user defined BC. You have examples in any of the test cases that use Manufactured Solutions (MS). 

The easiest trick to have access to dt would be to store it inside the subroutine with a SAVE fortran statement. In the first iteration, dt=t-0. Then, you update it every time you have a new "t" as input in the subroutine. As an alternative, Horses3d supports fixed time step. You can fix the time step in the control file and use it in your subroutine.

**Reply by TRPrasanna** [reply]
Thank you for the reply. 

Regarding the boundary conditions , is there a straightforward way to apply more than one user-defined BCs? For context, I need to set up user-defined BCs at inflow and outflow boundaries, but the boundary conditions at inflow and outflow are not the same. Would the recommended way be to use an if condition to determine whether the coordinate that is being passed to UserDefinedState1 belongs to a certain boundary? Or perhaps there is a way to define UserDefinedState2 (I don't believe the latter is possible yet , looking at the source code).

**Reply by loganoz** [reply]
Check lines 238-243 in UserDefinedBC.f90. I think you can modify that part of the code, along with the keyword "udf number" to allow for more user-defined BCs. Give at a try and let me know if that works.
